crates/subspace-farmer/src/disk_piece_cache.rs

```diff
@@ -87,12 +77,23 @@ impl FilePool {
             cursor: AtomicU8::new(0),
         })
     }
+
+    fn read(&self) -> &DirectIoFile {
+        let position = usize::from(self.cursor.fetch_add(1, Ordering::Relaxed));
+        &self.files[position % PIECES_READING_CONCURRENCY]
+    }
+
+    fn write(&self) -> &DirectIoFile {
+        // Always the same file or else overlapping writes will be corrupted due to
+        // read/modify/write internals, which are in turn caused by alignment requirements
+        &self.files[0]
```

Comment:
This is rather confusing despite the comment. Can you rephrase to explain writing API for a single file and reading API from different files?

Reply:
When you read it is fine to read concurrently, but due to alignment requirements when we do not write chunks of 512 bytes at a multiple of 512 bytes offset, we need to first read 512 bytes, then replace what we want to modify in those bytes with new contents and write it back to disk. We can't just write arbitrary sizes at arbitrary offsets with Direct IO, see the `DirectIoFile` implementation.